metrics/otel.go

```diff
@@ -33,89 +38,85 @@ func SetupOTelSDK(ctx context.Context) (func(context.Context) error, error) {
 		return err
 	}
 
-	// handleErr calls shutdown for cleanup and makes sure that all errors are returned.
-	handleErr := func(inErr error) {
-		err = errors.Join(inErr, shutdown(ctx))
-	}
-
-	// Set up propagator.
-	prop := newPropagator()
-	otel.SetTextMapPropagator(prop)
-
-	// Set up trace provider.
-	tracerProvider, err := newTracerProvider()
+	res, err := resource.New(ctx,
+		resource.WithAttributes(
+			attribute.String("service.name", "radiance"),
+			attribute.String("library.language", "go"),
+			attribute.String("service.version", app.Version),
+		),
+	)
 	if err != nil {
-		handleErr(err)
-		return shutdown, err
+		return shutdown, fmt.Errorf("failed to create resource: %w", err)
 	}
-	shutdownFuncs = append(shutdownFuncs, tracerProvider.Shutdown)
-	otel.SetTracerProvider(tracerProvider)
 
-	// Set up meter provider.
-	meterProvider, err := newMeterProvider()
-	if err != nil {
-		handleErr(err)
-		return shutdown, err
+	if cfg.TracesEnabled {
+		shutdownFunc, err := initTracer(res, cfg)
+		if err != nil {
+			return shutdown, fmt.Errorf("failed to initialize tracer: %w", err)
+		}
+		// Successfully initialized tracer
+		shutdownFuncs = append(shutdownFuncs, shutdownFunc)
+		slog.Info("OpenTelemetry tracer initialized")
 	}
-	shutdownFuncs = append(shutdownFuncs, meterProvider.Shutdown)
-	otel.SetMeterProvider(meterProvider)
 
-	// Set up logger provider.
-	loggerProvider, err := newLoggerProvider()
-	if err != nil {
-		handleErr(err)
-		return shutdown, err
+	if cfg.MetricsEnabled {
+		// Initialize the meter provider with the same gRPC connection.
+		// This is useful to avoid creating multiple connections to the same endpoint.
+		mp, err := initMeterProvider(ctx, res, cfg)
+		if err != nil {
+			return shutdown, fmt.Errorf("failed to initialize meter provider: %w", err)
+		}
+		// Successfully initialized meter provider
+		shutdownFuncs = append(shutdownFuncs, mp)
 	}
-	shutdownFuncs = append(shutdownFuncs, loggerProvider.Shutdown)
-	global.SetLoggerProvider(loggerProvider)
 
 	return shutdown, nil
 }
 
-func newPropagator() propagation.TextMapPropagator {
-	return propagation.NewCompositeTextMapPropagator(
-		propagation.TraceContext{},
-		propagation.Baggage{},
+func initTracer(res *resource.Resource, cfg common.OTEL) (func(context.Context) error, error) {
+	exporter, err := otlptrace.New(
+		context.Background(),
+		otlptracegrpc.NewClient(
+			otlptracegrpc.WithTLSCredentials(credentials.NewClientTLSFromCert(nil, "")),
```

Comment:
Since we're not providing any values to the TLS credentials I guess we can remove this option
```suggestion
```

Reply:
Pretty sure that's just the _client_ TLS credentials. Without that line I don't think it'll use TLS at all, but I'll check that.